src/main/java/transforms/HashTransform.java

```diff
@@ -81,13 +80,27 @@ public HashType getHashFunction() {
     }
 
     @Override
-    public Object apply( Map<String, String> row ) {
-        Hasher hasher = hf.newHasher();
-        for ( String s : columns ) {
-            if ( !StringUtils.isBlank( row.get( s ) ) ) {
-                hasher.putString( row.get( s ),  Charsets.UTF_8 );
+    public Object apply( Object input ) {
+        if (input instanceof Map){
+            Map<String, String> row = (Map) input;
+            Hasher hasher = hf.newHasher();
+            for ( String s : columns ) {
+                if ( !StringUtils.isBlank( row.get( s ) ) ) {
+                    hasher.putString( row.get( s ), Charsets.UTF_8 );
+                }
             }
+            return hasher.hash().toString();
+        } else if (input instanceof String) {
+            return super.apply( input );
+        } else {
+            throw  new IllegalArgumentException("This isn't a known input datatype");
```

Comment:
This is super minor, but you don't need `else` when you're returning within the previous `if` statement, so I think this pattern is a little easier to read:

```
if ( condition1 ) {
  ... logic ...
  return blah1;
}

if ( condition2 ) {
  return blah2;
}

throw new IllegalArgumentException("This isn't a known input datatype");
```